lucille-core/src/main/java/com/kmwllc/lucille/util/FileContentFetcher.java

```diff
@@ -0,0 +1,187 @@
+   */
+  public InputStream getInputStream(String path) throws IOException {
+    InputStream is;
+
+    if (path.startsWith("classpath:")) {
+      is = FileContentFetcher.class.getClassLoader().getResourceAsStream(path.substring(path.indexOf(":") + 1));
+    } else if (FileUtils.isValidURI(path)) {
+      URI pathURI = URI.create(path);
+      String activeClient = pathURI.getScheme() != null ? pathURI.getScheme() : "file";
+      StorageClient storageClient = availableClients.get(activeClient);
+
+      if (storageClient == null) {
+        throw new IOException("No Storage Client available to get content for path " + path + ".");
+      }
+
+      is = storageClient.getFileContentStream(pathURI);
+    } else {
+      is = new FileInputStream(path);
+    }
+
+    if (is == null) {
+      throw new IOException("No InputStream could be created for the path " + path + ".");
+    }
+
+    return is;
+  }
+
+  /**
+   * Returns a Reader for the file at the given path. If the path begins with "classpath:" the prefix will be removed
+   * and the file will be read from the classpath. Otherwise, it will be read from the local file system. Will use UTF-8
+   * encoding.
+   */
+  public Reader getReader(String path) throws IOException {
+    return getReader(path, "utf-8");
+  }
+
+  /**
+   * Returns a Reader for the file at the given path using the given encoding. If the path begins with "classpath:"
+   * the prefix will be removed and the file will be read from the classpath. Otherwise, it will be read from the local file system.
+   */
+  public Reader getReader(String path, String encoding) throws IOException {
+    InputStream stream = getInputStream(path);
+
+    // This method of creating the Reader is used because it handles non-UTF-8 characters by replacing them with UTF
+    // chars, rather than throwing an Exception.
+    // https://stackoverflow.com/questions/26268132/all-inclusive-charset-to-avoid-java-nio-charset-malformedinputexception-input
+    return new BufferedReader(new InputStreamReader(stream, encoding));
+  }
+
+  /**
+   * Counts the number of lines in the file at the given path.
+   */
+  public int countLines(String path) throws IOException {
+    try (BufferedReader reader = new BufferedReader(getReader(path))) {
+      int lines = 0;
+
+      while (reader.readLine() != null) {
+        lines++;
+      }
+
+      return lines;
+    }
+  }
+
+  // -- Convenience / Static Methods --
+  /**
+   * Returns an InputStream for the file at the given path, which could be in the classpath, a local file path, or a URI
+   * to a supported Storage service. An exception may occur - in this case, the StorageClient(s) created will be shutdown.
+   * If an InputStream is returned successfully, its close() method will shutdown the StorageClient(s) this method temporarily
+   * creates. Be sure to close the returned stream!
+   */
+  public static InputStream getSingleInputStream(String path, Config cloudOptions) throws IOException {
```

Comment:
look for a more explicit naming option

see if we can avoid initializing storage clients that aren't needed for accessing the given path